nxdrive/engine/engine.py

```diff
@@ -698,12 +698,16 @@ def cancel_upload(self, transfer_uid: int) -> None:
         upload = self.dao.get_dt_upload(uid=transfer_uid)
         if not upload:
             return
-        self.remote.cancel_batch(upload.batch)
-        self.dao.remove_transfer("upload", upload.path, is_direct_transfer=True)
-
         doc_pair = self.dao.get_state_from_local(upload.path)
         if not doc_pair:
             return
+        if upload.status is TransferStatus.ONGOING and doc_pair.processor:
```

Comment:
Could you add a line blank and a comment to explain how to get here and why this code is needed?